src/persist/src/s3.rs

```diff
@@ -380,14 +374,14 @@ impl Blob for S3Blob {
 
         // Get the remaining number of parts
         let num_parts = match first_part.parts_count() {
-            // For a non-multipart upload, parts_count will be 0. The rest of  the code works
+            // For a non-multipart upload, parts_count will be None. The rest of  the code works
             // perfectly well if we just pretend this was a multipart upload of 1 part.
-            0 => 1,
+            None => 1,
             // For any positive value greater than 0, just return it.
-            parts @ 1.. => parts,
-            // A negative value is invalid.
-            bad => {
-                assert!(bad < 0);
+            Some(parts @ 1..) => parts,
+            // A non-positive value is invalid.
+            Some(bad) => {
+                assert!(bad <= 0);
                 return Err(anyhow!("unexpected number of s3 object parts: {}", bad).into());
             }
         };
```

Comment:
@danhhz or @MaterializeInc/persist please scrutinize this change. The new SDK seems to report `None` rather than `0` for a non-multipart upload, and never report `Some(0)`. This seems strictly more correct than the old SDK behavior, but it's worth double checking my logic here. I'm assuming there's plenty of test coverage of this logic, but wanted to double check.

Reply:
@bkirwi Can you take this?